src/index.ts

```diff
@@ -1,105 +1,164 @@
+export const onProcessMetrics = (pid: number, options: onProcessMetricsOptions): Observable<PidUsage[]> => {
   options = { samplingInterval: 1000, ...options };
-  return getSharedProcessMetricsPoller(app, options.samplingInterval)
-}
-
+  return getSharedProcessMetricsPoller(pid, options.samplingInterval);
+};
 
 export interface ExtendedProcessMetric extends Electron.ProcessMetric {
   webContents?: {
-    type: string,
-    id: number,
-    pid: number,
-    URL: string,
-    URLDomain: string
-  }[]
+    type: string;
+    id: number;
+    pid: number;
+    URL: string;
+    URLDomain: string;
+  }[];
 }
 
+const getExtendedAppMetrics = (app: Electron.App) => {
+  const appMetrics = app.getAppMetrics();
+  const allWebContents = webContents.getAllWebContents();
+  const webContentsInfo = allWebContents.map((wc: overridenWebContents) => ({
+    type: wc.getType(),
+    id: wc.id,
+    pid: wc.getOSProcessId(),
+    URL: wc.getURL(),
+    URLDomain: extractURLDomain(wc.getURL()),
+  }));
+
+  return appMetrics.map(proc => {
+    const report: ExtendedProcessMetric = proc;
+
+    const wc = webContentsInfo.find(wc => wc.pid === proc.pid);
+    if (!wc) return report;
+
+    report.webContents = [wc];
+
+    return report;
+  });
+};
+
 /**
  * Returns an Rx.Observable that emits reports of `ExtendedProcessMetric`
  * every `options.samplingInterval` ms.
- * 
+ *
  * Default `options.samplingInterval` = 1000ms
- * 
+ *
  * Compared to `onProcessMetrics` it adds data on the `webContents` associated
  * to the given process.
- * 
+ *
  * @param app the electron app instance
  * @param options
  */
 export const onExtendedProcessMetrics = (app: Electron.App, options: onProcessMetricsOptions = {}) =>
-  onProcessMetrics(app, options)
-    .map(appMetrics => {
-      const allWebContents = webContents.getAllWebContents();
-      const webContentsInfo = allWebContents.map((wc: overridenWebContents) => ({
-        type: wc.getType(),
-        id: wc.id,
-        pid: wc.getOSProcessId(),
-        URL: wc.getURL(),
-        URLDomain: extractURLDomain(wc.getURL())
-      }));
-
-      return appMetrics.map(proc => {
-        const report = proc as ExtendedProcessMetric;
-
-        const wc = webContentsInfo.find(wc => wc.pid === proc.pid);
-        if (!wc) return report;
-
-        report.webContents = [];
-        report.webContents.push(wc);
-
-        return report
-    })
+  Observable.interval(options.samplingInterval || 1000).map(() => {
+    return getExtendedAppMetrics(app);
   });
 
 export interface onExcessiveCPUUsageOptions extends onProcessMetricsOptions {
   /**Number of samples to consider */
-  samplesCount?: number
+  samplesCount?: number;
   /**CPU usage percent minimum to consider a sample exceeds CPU usage */
-  percentCPUUsageThreshold?: number
+  percentCPUUsageThreshold?: number;
+}
+
+/**
+ * Will emit an array `PidUsage` when a process exceeds the
```

Comment:
Worth mentioning it monitor the CPU usage of the give process and its children. I have hard time figuring it out.

Reply:
Worth mentioning also explains the difference with the original `onExcessiveCPUUsage`.